src/jet_client.rs

```diff
@@ -144,224 +151,331 @@ impl Future for JetMsgReader {
-                        let mut candidate = Candidate::new_v1();
-                        candidate.set_state(CandidateState::Accepted);
-                        candidate.set_server_transport(self.transport.clone());
-                        association.add_candidate(candidate);
-
-                        jet_associations.insert(uuid, association);
-
-                        // Build response
-                        self.response_msg = Some(JetMessage::JetAcceptRsp(JetAcceptRsp {
-                            status_code: StatusCode::OK,
-                            version: request.version,
-                            association: uuid,
-                            instance: self.config.jet_instance(),
-                            timeout: ACCEPT_REQUEST_TIMEOUT_SEC,
-                        }));
-                    }
-                    2 => {
-                        let mut status_code = StatusCode::BAD_REQUEST;
-
-                        if let Some(association) = jet_associations.get_mut(&request.association) {
-                            if association.version() == JET_VERSION_V2 {
-                                if let Some(candidate) = association.get_candidate_mut(request.candidate) {
-                                    if candidate.transport_type() == TransportType::Tcp {
-                                        candidate.set_state(CandidateState::Accepted);
-                                        candidate.set_server_transport(self.transport.clone());
-                                        status_code = StatusCode::OK;
-                                    }
-                                } else {
-                                    status_code = StatusCode::NOT_FOUND;
+            let (status_code, association) = match self.request_msg.version {
+                1 => {
+                    // Candidate creation
+                    let mut candidate = Candidate::new_v1();
+                    candidate.set_state(CandidateState::Accepted);
+
+                    // Association creation
+                    let uuid = Uuid::new_v4();
+                    let mut association = Association::new(uuid, JET_VERSION_V1);
+                    association.add_candidate(candidate);
+                    self.association_uuid = Some(uuid.clone());
+
+                    jet_associations.insert(uuid, association);
+
+                    (StatusCode::OK, uuid)
+                }
+                2 => {
+                    let mut status_code = StatusCode::BAD_REQUEST;
+
+                    if let Some(association) = jet_associations.get_mut(&request.association) {
+                        if association.version() == JET_VERSION_V2 {
+                            if let Some(candidate) = association.get_candidate_mut(request.candidate) {
+                                if candidate.transport_type() == TransportType::Tcp {
+                                    candidate.set_state(CandidateState::Accepted);
```

Comment:
Here we do not set transport, because using the transport, we still need to respond to the JetAcceptResponse server. We will set it when we send the response to the server.